packages/bsky/src/api/app/bsky/graph/getKnownFollowers.ts

```diff
@@ -0,0 +1,122 @@
+import { InvalidRequestError } from '@atproto/xrpc-server'
+import { Server } from '../../../../lexicon'
+import { QueryParams } from '../../../../lexicon/types/app/bsky/graph/getKnownFollowers'
+import AppContext from '../../../../context'
+import {
+  HydrationFnInput,
+  PresentationFnInput,
+  RulesFnInput,
+  SkeletonFnInput,
+  createPipeline,
+} from '../../../../pipeline'
+import { HydrateCtx, Hydrator } from '../../../../hydration/hydrator'
+import { Views } from '../../../../views'
+import { clearlyBadCursor, resHeaders } from '../../../util'
+
+export default function (server: Server, ctx: AppContext) {
+  const getKnownFollowers = createPipeline(
+    skeleton,
+    hydration,
+    noBlocks,
+    presentation,
+  )
+  server.app.bsky.graph.getKnownFollowers({
+    auth: ctx.authVerifier.standard,
+    handler: async ({ params, auth, req }) => {
+      const viewer = auth.credentials.iss
+      const labelers = ctx.reqLabelers(req)
+      const hydrateCtx = await ctx.hydrator.createContext({
+        labelers,
+        viewer,
+      })
+
+      const result = await getKnownFollowers(
+        { ...params, hydrateCtx: hydrateCtx.copy({ viewer }) },
+        ctx,
+      )
+
+      return {
+        encoding: 'application/json',
+        body: result,
+        headers: resHeaders({ labelers: hydrateCtx.labelers }),
+      }
+    },
+  })
+}
+
+const skeleton = async (input: SkeletonFnInput<Context, Params>) => {
+  const { params, ctx } = input
+  const [subjectDid] = await ctx.hydrator.actor.getDidsDefined([params.actor])
+  if (!subjectDid) {
+    throw new InvalidRequestError(`Actor not found: ${params.actor}`)
+  }
+  if (clearlyBadCursor(params.cursor)) {
+    return { subjectDid, knownFollowers: [], cursor: undefined }
+  }
+  const { knownFollowers } = await ctx.hydrator.graph.getKnownFollowers({
+    viewerDid: params.hydrateCtx.viewer,
+    subjectDid,
+    cursor: params.cursor,
+    // default here since no dataplane support
+    limit: params.limit || 50,
+  })
+  return {
+    subjectDid,
+    knownFollowers,
+    cursor: undefined,
+  }
+}
+
+const hydration = async (
+  input: HydrationFnInput<Context, Params, SkeletonState>,
+) => {
+  const { ctx, params, skeleton } = input
+  const { knownFollowers } = skeleton
+  const profilesState = await ctx.hydrator.hydrateProfiles(
+    knownFollowers.concat(skeleton.subjectDid),
+    params.hydrateCtx,
+  )
+  return profilesState
+}
+
+const noBlocks = (input: RulesFnInput<Context, Params, SkeletonState>) => {
+  const { skeleton, hydration, ctx } = input
+  skeleton.knownFollowers = skeleton.knownFollowers.filter((did) => {
+    return !ctx.views.viewerBlockExists(did, hydration)
+  })
+  return skeleton
+}
+
+const presentation = (
+  input: PresentationFnInput<Context, Params, SkeletonState>,
+) => {
+  const { ctx, hydration, skeleton } = input
+  const { knownFollowers } = skeleton
+  const isNoHosted = (did: string) => ctx.views.actorIsNoHosted(did, hydration)
```

Comment:
since we never include takedowns in hydration (`includeTakedowns` on `HydrateCtx`), it's actually not required to do this check, they get stripped out ahead of time 👍

Reply:
oh excellent 👍